spec/iot-js-spec.md

```diff
@@ -123,31 +118,29 @@ dictionary OicDeviceInfo {
 interface OicClient: EventTarget {
   // client API: discovery; Promise succeeds when request is successfully sent
   Promise<void> findResources(OicDiscoveryOptions options);  // GET oic/res
-  Promise<void> findDevices(OicDiscoveryOptions options);  // GET /oic/d
+  Promise<void> findDevices();  // GET /oic/d
 
   // client API: CRUDN
-  Promise<OicResource> createResource(OicResourceInit resource);  // create remote res
-  Promise<OicResource> retrieveResource(USVString resourceId);  // get remote res
-  Promise<void> updateResource(USVString resourceId, OicResourceInit resource);  // post or put remote res
-  Promise<void> deleteResource(USVString resourceId);  // delete remote res
-  Promise<OicResource> startObserving(USVString resourceId);  // retrieve + observe flag
-  Promise<void> cancelObserving(USVString id);
-
-  attribute EventHandler<OicResourceChangeEvent> onresourcechange;  // observe
```

Comment:
node.js has an [EventEmitter](https://nodejs.org/api/events.html#events_event_newlistener) API which lets me know whenever a new listener is attached to an object, and how many listeners are currently attach. That lets me do special things like initiating observation when someone attaches the first handler and cancelling observation when someone detaches the last handler. Wonderful! However, you gotta ask yourself if Tizen and Crosswalk also have this capability.